File: function-aws/features/function-aws-java/skeleton/src/main/java/@defaultPackage.path@/@project.className@Function.java
package @defaultPackage@;

import io.micronaut.function.executor.FunctionInitializer;
import io.micronaut.function.FunctionBean;
import javax.inject.*;
import java.io.IOException;
import java.util.function.Function;

@FunctionBean("@project.name@")
public class @project.className@Function extends FunctionInitializer implements Function<@project.className@, @project.className@> {

    @Override
    public @project.className@ apply(@project.className@ msg) {
         return msg;
    }

    /**
     * This main method allows running the function as a CLI application using: echo '{}' | java -jar function.jar 
     * where the argument to echo is the JSON to be parsed.
     */
    public static void main(String...args) throws IOException {
        @project.className@Function function = new @project.className@Function();
        function.run(args, (context)-> function.apply(context.get(@project.className@.class)));
    }    
}

File: function-aws/features/function-aws-kotlin/skeleton/src/main/kotlin/@defaultPackage.path@/@project.className@Function.kt
package @defaultPackage@;

import io.micronaut.function.executor.FunctionInitializer
import io.micronaut.function.FunctionBean;
import javax.inject.*;
import java.util.function.Function;

@FunctionBean("@project.name@")
class @project.className@Function : FunctionInitializer(), Function<@project.className@, @project.className@> {

    override fun apply(msg : @project.className@) : @project.className@ {
         return msg
    }   
}

/**
 * This main method allows running the function as a CLI application using: echo '{}' | java -jar function.jar 
 * where the argument to echo is the JSON to be parsed.
 */
fun main(args : Array<String>) { 
    val function = @project.className@Function()
    function.run(args, { context -> function.apply(context.get(@project.className@::class.java))})
}